Fix PEP8 errors when running under Python 3

Probably we need to ignore 402.

``` sh
$ make pep8
./bin/filter-fasta-by-complexity.py:50:9: E731 do not assign a lambda expression, use a def
./bin/filter-fasta-by-taxonomy.py:77:9: E731 do not assign a lambda expression, use a def
./bin/filter-fasta-by-taxonomy.py:81:9: E731 do not assign a lambda expression, use a def
./bin/noninteractive-alignment-panel.py:23:1: E402 module level import not at top of file
./bin/noninteractive-alignment-panel.py:24:1: E402 module level import not at top of file
./dark/alignments.py:364:33: W503 line break before binary operator
./dark/blast/params.py:33:21: W503 line break before binary operator
./dark/graphics.py:207:9: E731 do not assign a lambda expression, use a def
./dark/titles.py:177:13: E731 do not assign a lambda expression, use a def
./dark/utils.py:51:1: E402 module level import not at top of file
./test/blast/test_alignments.py:392:9: E731 do not assign a lambda expression, use a def
./test/blast/test_alignments.py:418:9: E731 do not assign a lambda expression, use a def
./test/blast/test_alignments.py:1244:9: E731 do not assign a lambda expression, use a def
./test/diamond/test_alignments.py:343:9: E731 do not assign a lambda expression, use a def
./test/diamond/test_alignments.py:370:9: E731 do not assign a lambda expression, use a def
./test/diamond/test_alignments.py:1250:9: E731 do not assign a lambda expression, use a def
./test/test_features.py:156:9: E731 do not assign a lambda expression, use a def
./test/test_features.py:166:9: E731 do not assign a lambda expression, use a def
./test/test_features.py:189:9: E731 do not assign a lambda expression, use a def
./test/test_features.py:199:9: E731 do not assign a lambda expression, use a def
./test/test_features.py:329:9: E731 do not assign a lambda expression, use a def
./test/test_features.py:342:9: E731 do not assign a lambda expression, use a def
./test/test_features.py:356:9: E731 do not assign a lambda expression, use a def
./test/test_features.py:374:9: E731 do not assign a lambda expression, use a def
./test/test_features.py:578:9: E731 do not assign a lambda expression, use a def
make: *** [pep8] Error 1
```

``` sh
$ pep8 --version
1.7.0
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,7 @@ install:
   pip install -r requirements.txt
 script:
   - pytest
+  - find bin dark test -name '*.py' -print0 | xargs -0 flake8 --ignore E402,W504
 notifications:
   irc: "irc.freenode.net#acorg"
   slack: acorg:PBqi8M1O5JVt26evSkRoX40z

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-.PHONY: check, tcheck, pycodestyle, pyflakes, lint, wc, clean, clobber, upload
+.PHONY: check, tcheck, pycodestyle, pyflakes, flake8, lint, wc, clean, clobber, upload
 
 XARGS := xargs $(shell test $$(uname) = Linux && echo -r)
 
@@ -8,13 +8,17 @@ check:
 tcheck:
 	env PYTHONPATH=. trial --rterrors test
 
+pytest:
+	pytest
+
 pycodestyle:
-	find . -path './.tox' -prune -o -path './build' -prune -o -path './dist' -prune -o -name '*.py' -print0 | $(XARGS) -0 pycodestyle
+	find bin dark test -name '*.py' -print0 | $(XARGS) -0 pycodestyle --ignore E402,W504
 
 pyflakes:
-	find .  -path './.tox' -prune -path './build' -prune -o -path './dist' -prune -o -name '*.py' -print0 | $(XARGS) -0 pyflakes
+	find bin dark test -name '*.py' -print0 | $(XARGS) -0 pyflakes
 
-lint: pycodestyle pyflakes
+flake8:
+	find bin dark test -name '*.py' -print0 | $(XARGS) -0 flake8 --ignore E402,W504
 
 wc:
 	find . -path './.tox' -prune -o -path './build' -prune -o -path './dist' -prune -o -name '*.py' -print0 | $(XARGS) -0 wc -l

--- a/bin/filter-fasta-by-taxonomy.py
+++ b/bin/filter-fasta-by-taxonomy.py
@@ -74,11 +74,14 @@ if __name__ == '__main__':
 
     if args.detailsFile is not None:
         detailsFp = open(args.detailsFile, 'w')
-        details = lambda accept, readId, taxonomy: (
-            writeDetails(accept, readId, taxonomy, detailsFp))
+
+        def details(accept, readId, taxonomy):
+            return writeDetails(accept, readId, taxonomy, detailsFp)
     else:
         detailsFp = None
-        details = lambda accept, readId, taxonomy: None
+
+        def details(accept, readId, taxonomy):
+            return None
 
     lineageFetcher = LineageFetcher()
     reads = FastaReads(sys.stdin)

--- a/bin/get-reads.py
+++ b/bin/get-reads.py
@@ -45,6 +45,7 @@ def main(recordFilenames, fastaFilename, title, xRange, bitRange):
                        titleAlignment.read.id, hsp.subjectStart,
                        hsp.subjectEnd, hsp.score.score)))
 
+
 if __name__ == '__main__':
     if len(sys.argv) < 3:
         print((

--- a/bin/pre-commit.sh
+++ b/bin/pre-commit.sh
@@ -13,9 +13,6 @@ then
     PATH="$VIRTUAL_ENV/bin:$PATH"
 fi
 
-tmp=/tmp/git-pre-commit-$$
-trap "rm -f $tmp" 0 1 2 3 15
-
 make flake8
 
 if [ $? -ne 0 ]
@@ -32,17 +29,20 @@ then
     exit 1
 fi
 
+# tmp=/tmp/git-pre-commit-$$
+# trap "rm -f $tmp" 0 1 2 3 15
+#
 # make flake8 > $tmp 2>&1
-
+#
 # if [ $? -ne 0 ]
 # then
 #     echo 'COMMIT FAILED: make flake8 did not run cleanly:' >&2
 #     cat $tmp >&2
 #     exit 1
 # fi
-
+#
 # make pytest > $tmp 2>&1
-
+#
 # if [ $? -ne 0 ]
 # then
 #     echo 'COMMIT FAILED: make check did not run cleanly:' >&2

--- a/bin/pre-commit.sh
+++ b/bin/pre-commit.sh
@@ -16,20 +16,36 @@ fi
 tmp=/tmp/git-pre-commit-$$
 trap "rm -f $tmp" 0 1 2 3 15
 
-make lint > $tmp 2>&1
+make flake8
 
 if [ $? -ne 0 ]
 then
-    echo 'COMMIT FAILED: make lint did not run cleanly:' >&2
-    cat $tmp >&2
+    echo 'COMMIT FAILED: make flake8 did not run cleanly:' >&2
     exit 1
 fi
 
-make check > $tmp 2>&1
+make pytest
 
 if [ $? -ne 0 ]
 then
-    echo 'COMMIT FAILED: make check did not run cleanly:' >&2
-    cat $tmp >&2
+    echo 'COMMIT FAILED: make pytest did not run cleanly:' >&2
     exit 1
 fi
+
+# make flake8 > $tmp 2>&1
+
+# if [ $? -ne 0 ]
+# then
+#     echo 'COMMIT FAILED: make flake8 did not run cleanly:' >&2
+#     cat $tmp >&2
+#     exit 1
+# fi
+
+# make pytest > $tmp 2>&1
+
+# if [ $? -ne 0 ]
+# then
+#     echo 'COMMIT FAILED: make check did not run cleanly:' >&2
+#     cat $tmp >&2
+#     exit 1
+# fi

--- a/bin/proteins-to-pathogens.py
+++ b/bin/proteins-to-pathogens.py
@@ -24,7 +24,7 @@ output.  Alternately, you can also provide file names on the command line.
 Typical usage:
 
   $ find . -name summary-proteins | proteins-to-pathogens.py \
-        --sampleNameRegex '(Sample_\d+)/' --html > index.html
+        --sampleNameRegex "(Sample_\\d+)/" --html > index.html
 
 Input files must contain lines in the following format:
 

--- a/bin/sam-to-fasta-alignment.py
+++ b/bin/sam-to-fasta-alignment.py
@@ -20,81 +20,81 @@ parser.add_argument(
     'samFile', help='The SAM/BAM file to read.')
 
 parser.add_argument(
-     '--minLength', type=int, default=0,
-     help='Do not output anything for queries shorter than this.')
+    '--minLength', type=int, default=0,
+    help='Do not output anything for queries shorter than this.')
 
 parser.add_argument(
-     '--referenceName',
-     help=('The name of the reference sequence to print alignments for. '
-           'This is only needed if the SAM/BAM alignment was made against '
-           'multiple references *and* they are not all of the same length. '
-           'If there is only one reference sequence or if all reference '
-           'sequences are of the same length, there is no need to provide a '
-           'reference name.'))
+    '--referenceName',
+    help=('The name of the reference sequence to print alignments for. '
+          'This is only needed if the SAM/BAM alignment was made against '
+          'multiple references *and* they are not all of the same length. '
+          'If there is only one reference sequence or if all reference '
+          'sequences are of the same length, there is no need to provide a '
+          'reference name.'))
 
 parser.add_argument(
-     '--rcSuffix', default='',
-     help=('A string to add to the end of query names that are reverse '
-           'complemented. This is added before the /1, /2, etc., that are '
-           'added for duplicated ids (if there are duplicates and '
-           '--allowDuplicateIds is not used)'))
+    '--rcSuffix', default='',
+    help=('A string to add to the end of query names that are reverse '
+          'complemented. This is added before the /1, /2, etc., that are '
+          'added for duplicated ids (if there are duplicates and '
+          '--allowDuplicateIds is not used)'))
 
 parser.add_argument(
-     '--dropSecondary', default=False, action='store_true',
-     help='If given, secondary matches will not be output.')
+    '--dropSecondary', default=False, action='store_true',
+    help='If given, secondary matches will not be output.')
 
 parser.add_argument(
-     '--dropSupplementary', default=False, action='store_true',
-     help='If given, supplementary matches will not be output.')
+    '--dropSupplementary', default=False, action='store_true',
+    help='If given, supplementary matches will not be output.')
 
 parser.add_argument(
-     '--dropDuplicates', default=False, action='store_true',
-     help=('If given, matches flagged as optical or PCR duplicates will '
-           'not be output.'))
+    '--dropDuplicates', default=False, action='store_true',
+    help=('If given, matches flagged as optical or PCR duplicates will '
+          'not be output.'))
 
 parser.add_argument(
-     '--allowDuplicateIds', default=False, action='store_true',
-     help=('If given, repeated query ids (due to secondary or supplemental '
-           'matches) will not have /1, /2, etc. appended to their ids. So '
-           'repeated ids may appear in the output FASTA.'))
+    '--allowDuplicateIds', default=False, action='store_true',
+    help=('If given, repeated query ids (due to secondary or supplemental '
+          'matches) will not have /1, /2, etc. appended to their ids. So '
+          'repeated ids may appear in the output FASTA.'))
 
 parser.add_argument(
-     '--keepQCFailures', default=False, action='store_true',
-     help=('If given, reads that are considered quality control failures will '
-           'be included in the output.'))
+    '--keepQCFailures', default=False, action='store_true',
+    help=('If given, reads that are considered quality control failures will '
+          'be included in the output.'))
 
 parser.add_argument(
-     '--listReferenceNames', default=False, action='store_true',
-     help=('If given, the names of all reference sequences will be '
-           'printed to standard output and the program will exit.'))
+    '--listReferenceNames', default=False, action='store_true',
+    help=('If given, the names of all reference sequences will be '
+          'printed to standard output and the program will exit.'))
 
 parser.add_argument(
-     '--listReferenceInsertions', default=False, action='store_true',
-     help=('If given, information about reference sequence insertions will be '
-           'printed to standard error. These correspond to "I" CIGAR '
-           'operations that for the match would require inserting query bases '
-           'into the reference. Because we cannot change the reference (in '
-           'fact we typically do not have the reference in the SAM/BAM file), '
-           'we cut the inserted bases out of the aligned query and save the '
-           'information about what would have been inserted and where. That '
-           'information is printed by this option. The output gives the '
-           '0-based offset where the inserted base would be placed, followed '
-           'by a list of the nucleotides that were suggested as being '
-           'inserted and the number of times each nucleotide was suggested. '
-           'So for example the output might contain "27: T:3, G:10" which '
-           'indicates that 13 query (3 with T and 10 with G) matches would '
-           'insert a nucleotide into the reference at offset 27.'))
+    '--listReferenceInsertions', default=False, action='store_true',
+    help=('If given, information about reference sequence insertions will be '
+          'printed to standard error. These correspond to "I" CIGAR '
+          'operations that for the match would require inserting query bases '
+          'into the reference. Because we cannot change the reference (in '
+          'fact we typically do not have the reference in the SAM/BAM file), '
+          'we cut the inserted bases out of the aligned query and save the '
+          'information about what would have been inserted and where. That '
+          'information is printed by this option. The output gives the '
+          '0-based offset where the inserted base would be placed, followed '
+          'by a list of the nucleotides that were suggested as being '
+          'inserted and the number of times each nucleotide was suggested. '
+          'So for example the output might contain "27: T:3, G:10" which '
+          'indicates that 13 query (3 with T and 10 with G) matches would '
+          'insert a nucleotide into the reference at offset 27.'))
 
 parser.add_argument(
-     '--rcNeeded', default=False, action='store_true',
-     help=('If given, queries that are flagged as matching when reverse '
-           'complemented will be reverse complemented in the output. This '
-           'must be used if the program that created the SAM/BAM input '
-           'flags reversed matches but does not also store the reverse '
-           'complemented query. The bwa program (mem and aln followed by '
-           'samse) stores the queries reversed complemented if the match '
-           'was, so this option is not needed for bwa. If in doubt, test the '
-           'output of your matching program as this is very important!'))
+    '--rcNeeded', default=False, action='store_true',
+    help=('If given, queries that are flagged as matching when reverse '
+          'complemented will be reverse complemented in the output. This '
+          'must be used if the program that created the SAM/BAM input '
+          'flags reversed matches but does not also store the reverse '
+          'complemented query. The bwa program (mem and aln followed by '
+          'samse) stores the queries reversed complemented if the match '
+          'was, so this option is not needed for bwa. If in doubt, test the '
+          'output of your matching program as this is very important!'))
 
 
 def baseCountsToStr(counts):

--- a/dark/blast/alignments.py
+++ b/dark/blast/alignments.py
@@ -49,7 +49,7 @@ class BlastReadsAlignments(ReadsAlignments):
         files, or if BLAST parameters in all files do not match.
     """
 
-    def __init__(self, reads, blastFilenames,  databaseFilename=None,
+    def __init__(self, reads, blastFilenames, databaseFilename=None,
                  databaseDirectory=None, sqliteDatabaseFilename=None,
                  scoreClass=HigherIsBetterScore,
                  sortBlastFilenames=True, randomizeZeroEValues=True):

--- a/dark/blast/params.py
+++ b/dark/blast/params.py
@@ -29,8 +29,8 @@ def checkCompatibleParams(initialParams, laterParams):
     err = []
     for param in initialParams:
         if param in laterParams:
-            if (param not in VARIABLE_PARAMS
-                    and initialParams[param] != laterParams[param]):
+            if (param not in VARIABLE_PARAMS and
+                    initialParams[param] != laterParams[param]):
                 err.append(
                     '\tParam %r initial value %r differs from '
                     'later value %r' % (param, initialParams[param],

--- a/dark/dna.py
+++ b/dark/dna.py
@@ -31,6 +31,7 @@ AMBIGUOUS = {
 BASES_TO_AMBIGUOUS = dict(
     (''.join(sorted(bases)), symbol) for symbol, bases in AMBIGUOUS.items())
 
+
 def compareDNAReads(read1, read2, matchAmbiguous=True, gapChars=('-'),
                     offsets=None):
     """
@@ -48,9 +49,9 @@ def compareDNAReads(read1, read2, matchAmbiguous=True, gapChars=('-'),
     @param read1: A C{Read} instance or an instance of one of its subclasses.
     @param read2: A C{Read} instance or an instance of one of its subclasses.
     @param matchAmbiguous: If C{True}, count ambiguous nucleotides that are
-        possibly correct as actually being correct, and score these in the 
-        ambiguousMatchCount. Otherwise, we are strict and insist that only 
-        non-ambiguous nucleotides can contribute to the matching nucleotide 
+        possibly correct as actually being correct, and score these in the
+        ambiguousMatchCount. Otherwise, we are strict and insist that only
+        non-ambiguous nucleotides can contribute to the matching nucleotide
         count.
     @param gapChars: An iterable containing characters that should be
         considered to be gaps.
@@ -73,11 +74,11 @@ def compareDNAReads(read1, read2, matchAmbiguous=True, gapChars=('-'),
 
     def _ambiguousMatch(a, b, matchAmbiguous):
         """
-        Checks if two characters match ambiguously if matchAmbiguous is True. 
-        A match is an ambiguous match if it is not an identical match, but the 
+        Checks if two characters match ambiguously if matchAmbiguous is True.
+        A match is an ambiguous match if it is not an identical match, but the
         sets of ambiguous characters overlap.
         """
-        return (matchAmbiguous and 
+        return (matchAmbiguous and
                 not _identicalMatch(a, b) and
                 AMBIGUOUS.get(a, empty) & AMBIGUOUS.get(b, empty))
 

--- a/dark/graphics.py
+++ b/dark/graphics.py
@@ -650,8 +650,8 @@ def alignmentPanelHTML(titlesAlignments, sortOn='maxScore',
         sequences.
     @param showOrfs: If C{True}, open reading frames will be displayed.
     @raise TypeError: If C{outputDir} is C{None}.
-    @raise ValueError: If C{outputDir} is None or exists but is not a directory or if
-        C{xRange} is not "subject" or "reads".
+    @raise ValueError: If C{outputDir} is None or exists but is not a
+        directory or if C{xRange} is not "subject" or "reads".
     """
 
     if xRange not in ('subject', 'reads'):

--- a/dark/local_align.py
+++ b/dark/local_align.py
@@ -79,41 +79,41 @@ class LocalAlignment(object):
         max_row = 0
         max_col = 0
 
-        for row in range(1, len(self.seq2Seq)+1):
-            for col in range(1, len(self.seq1Seq)+1):
+        for row in range(1, len(self.seq2Seq) + 1):
+            for col in range(1, len(self.seq1Seq) + 1):
                 # Calculate match score
-                letter1 = self.seq1Seq[col-1]
-                letter2 = self.seq2Seq[row-1]
+                letter1 = self.seq1Seq[col - 1]
+                letter2 = self.seq2Seq[row - 1]
                 if letter1 == letter2:
-                    diagonal_score = (table[row-1][col-1]['score'] +
+                    diagonal_score = (table[row - 1][col - 1]['score'] +
                                       self.match)
                 else:
-                    diagonal_score = (table[row-1][col-1]['score'] +
+                    diagonal_score = (table[row - 1][col - 1]['score'] +
                                       self.mismatch)
 
-                ins_run = table[row-1][col]['ins']
-                del_run = table[row][col-1]['del']
+                ins_run = table[row - 1][col]['ins']
+                del_run = table[row][col - 1]['del']
 
                 # Calculate gap scores ensuring extension is not > 0
-                if table[row-1][col]['ins'] <= 0:
-                    ins_score = table[row-1][col]['score'] + self.gapOpen
+                if table[row - 1][col]['ins'] <= 0:
+                    ins_score = table[row - 1][col]['score'] + self.gapOpen
                 else:
                     if self.gapExtend + ins_run * self.gapExtendDecay <= 0.0:
-                        ins_score = (table[row-1][col]['score'] +
+                        ins_score = (table[row - 1][col]['score'] +
                                      self.gapExtend +
                                      ins_run * self.gapExtendDecay)
                     else:
-                        ins_score = table[row-1][col]['score']
+                        ins_score = table[row - 1][col]['score']
 
-                if table[row-1][col]['del'] <= 0:
-                    del_score = table[row][col-1]['score'] + self.gapOpen
+                if table[row - 1][col]['del'] <= 0:
+                    del_score = table[row][col - 1]['score'] + self.gapOpen
                 else:
                     if self.gapExtend + del_run * self.gapExtendDecay <= 0.0:
-                        del_score = (table[row][col-1]['score'] +
+                        del_score = (table[row][col - 1]['score'] +
                                      self.gapExtend +
                                      del_run * self.gapExtendDecay)
                     else:
-                        del_score = table[row][col-1]['score']
+                        del_score = table[row][col - 1]['score']
 
                 # Choose best score
                 if diagonal_score <= 0 and ins_score <= 0 and del_score <= 0:
@@ -163,22 +163,23 @@ class LocalAlignment(object):
                 min_col = current_col + 1
                 break
             elif arrow == 'diagonal':
-                align1 += self.seq1Seq[current_col-1]
-                align2 += self.seq2Seq[current_row-1]
-                if self.seq1Seq[current_col-1] == self.seq2Seq[current_row-1]:
+                align1 += self.seq1Seq[current_col - 1]
+                align2 += self.seq2Seq[current_row - 1]
+                if self.seq1Seq[current_col - 1] == self.seq2Seq[
+                        current_row - 1]:
                     align += '|'
                 else:
                     align += ' '
                 current_row -= 1
                 current_col -= 1
             elif arrow == 'del':
-                align1 += self.seq1Seq[current_col-1]
+                align1 += self.seq1Seq[current_col - 1]
                 align2 += '-'
                 align += ' '
                 current_col -= 1
             elif arrow == 'ins':
                 align1 += '-'
-                align2 += self.seq2Seq[current_row-1]
+                align2 += self.seq2Seq[current_row - 1]
                 align += ' '
                 current_row -= 1
             else:

--- a/dark/mutations.py
+++ b/dark/mutations.py
@@ -306,7 +306,7 @@ def makeFrequencyGraph(allFreqs, title, substitution, pattern,
     index = 0
     data = []
     for item in patterns:
-        newData = toPlot[patterns[index]]/divisor
+        newData = toPlot[patterns[index]] / divisor
         data.append(newData)
         index += 1
     # create the bars
@@ -318,10 +318,10 @@ def makeFrequencyGraph(allFreqs, title, substitution, pattern,
         ax.set_title('%s \n %s' % (title, substitution), fontsize=20)
         ax.set_ylim(0, maxY)
         ax.set_ylabel('Absolute Number of Mutations', fontsize=16)
-        ax.set_xticks(ind+width)
+        ax.set_xticks(ind + width)
         ax.set_xticklabels(patterns, rotation=45, fontsize=8)
     if createFigure is False:
-        ax.set_xticks(ind+width)
+        ax.set_xticks(ind + width)
         ax.set_xticklabels(patterns, rotation=45, fontsize=0)
     else:
         if showFigure:
@@ -380,7 +380,7 @@ def makeFrequencyPanel(allFreqs, patientName):
             # add xAxis tick labels
             if i == 0:
                 ax[i][index].set_title(substitution, fontsize=13)
-            if i == len(allFreqs)-1 or i == (len(allFreqs)-1)/2:
+            if i == len(allFreqs) - 1 or i == (len(allFreqs) - 1) / 2:
                 if index < 3:
                     pat = ['ACA', 'ACC', 'ACG', 'ACT', 'CCA', 'CCC', 'CCG',
                            'CCT', 'GCA', 'GCC', 'GCG', 'GCT', 'TCA', 'TCC',

--- a/dark/proteins.py
+++ b/dark/proteins.py
@@ -37,7 +37,7 @@ from dark.reads import Reads
 #
 # I decided not to worry about nested [...] sections (there are only 2
 # instances that I know of).
-_PATHOGEN_RE = re.compile('^(.*)\[([^\]]+)\]$')
+_PATHOGEN_RE = re.compile(r'^(.*)\[([^\]]+)\]$')
 
 # The pathogen name assigned to proteins whose id strings cannot be parsed
 # for a pathogen name (see previous comment).  Do not use '<', '>' or any

--- a/dark/titles.py
+++ b/dark/titles.py
@@ -201,7 +201,8 @@ class TitleAlignments(list):
             and the count of that residue at that location as values.
         """
         if convertCaseTo == 'none':
-            convert = lambda x: x
+            def convert(x):
+                return x
         elif convertCaseTo == 'lower':
             convert = str.lower
         elif convertCaseTo == 'upper':

--- a/dark/utils.py
+++ b/dark/utils.py
@@ -7,6 +7,7 @@ import gzip
 from os.path import basename
 from contextlib import contextmanager
 from re import compile
+import numpy as np
 
 
 def numericallySortFilenames(names):
@@ -49,7 +50,6 @@ def numericallySortFilenames(names):
 # If numpy seems ok, use it. Else try for the Python 3.4 median function, else
 # write our own.
 
-import numpy as np
 _a = np.array([1])
 
 try:
@@ -59,20 +59,20 @@ except AttributeError:
     try:
         from statistics import median as _median
     except ImportError:
-        def _median(l):
+        def _median(numbers):
             """
             Find the median of a set of numbers.
 
-            @param l: A C{list} of numeric values.
-            @return: The median value in C{l}.
+            @param numbers: A C{list} of numeric values.
+            @return: The median value in C{numbers}.
             """
-            l = sorted(l)
-            n = len(l)
+            numbers = sorted(numbers)
+            n = len(numbers)
             if n % 2:
-                return l[n // 2]
+                return numbers[n // 2]
             else:
                 n //= 2
-                return (l[n - 1] + l[n]) / 2
+                return (numbers[n - 1] + numbers[n]) / 2
 else:
     # We have a working numpy. Just make sure we raise on an empty list as
     # numpy returns numpy.nan

--- a/requirements.txt
+++ b/requirements.txt
@@ -12,3 +12,4 @@ simplejson>=3.5.3
 six>=1.11.0
 pysam>=0.14.1
 mysql-connector-python>=8.0.11
+flake8>=3.5.0

--- a/test/blast/test_alignments.py
+++ b/test/blast/test_alignments.py
@@ -23,7 +23,7 @@ from dark.reads import Read, Reads, DNARead
 from dark.hsp import HSP, LSP
 from dark.score import LowerIsBetterScore
 from dark.blast.alignments import (
-    BlastReadsAlignments,  ZERO_EVALUE_UPPER_RANDOM_INCREMENT)
+    BlastReadsAlignments, ZERO_EVALUE_UPPER_RANDOM_INCREMENT)
 from dark.titles import TitlesAlignments
 from dark import ncbidb
 
@@ -550,10 +550,13 @@ class TestBlastReadsAlignments(TestCase):
         The adjustHspsForPlotting function must alter HSPs so that non-zero
         evalues are converted to the positive value of their negative exponent.
         """
-        result = lambda a, **kwargs: File([
-            dumps(PARAMS) + '\n', dumps(deepcopy(RECORD0)) + '\n',
-            dumps(deepcopy(RECORD1)) + '\n', dumps(deepcopy(RECORD2)) + '\n',
-            dumps(deepcopy(RECORD3)) + '\n'])
+        def result(a, **kwargs):
+            return File([
+                dumps(PARAMS) + '\n',
+                dumps(deepcopy(RECORD0)) + '\n',
+                dumps(deepcopy(RECORD1)) + '\n',
+                dumps(deepcopy(RECORD2)) + '\n',
+                dumps(deepcopy(RECORD3)) + '\n'])
 
         with patch.object(builtins, 'open') as mockMethod:
             mockMethod.side_effect = result
@@ -576,10 +579,14 @@ class TestBlastReadsAlignments(TestCase):
         The adjustHspsForPlotting function must alter HSPs so that zero
         evalues are set randomly high.
         """
-        result = lambda a, **kwargs: File([
-            dumps(PARAMS) + '\n', dumps(deepcopy(RECORD0)) + '\n',
-            dumps(deepcopy(RECORD1)) + '\n', dumps(deepcopy(RECORD2)) + '\n',
-            dumps(deepcopy(RECORD3)) + '\n', dumps(deepcopy(RECORD4)) + '\n'])
+        def result(a, **kwargs):
+            return File([
+                dumps(PARAMS) + '\n',
+                dumps(deepcopy(RECORD0)) + '\n',
+                dumps(deepcopy(RECORD1)) + '\n',
+                dumps(deepcopy(RECORD2)) + '\n',
+                dumps(deepcopy(RECORD3)) + '\n',
+                dumps(deepcopy(RECORD4)) + '\n'])
 
         with patch.object(builtins, 'open') as mockMethod:
             mockMethod.side_effect = result

--- a/test/diamond/test_alignments.py
+++ b/test/diamond/test_alignments.py
@@ -339,10 +339,13 @@ class TestDiamondReadsAlignments(TestCase):
         The adjustHspsForPlotting function must alter HSPs so that non-zero
         evalues are converted to the positive value of their negative exponent.
         """
-        result = lambda a: File([
-            dumps(PARAMS) + '\n', dumps(deepcopy(RECORD0)) + '\n',
-            dumps(deepcopy(RECORD1)) + '\n', dumps(deepcopy(RECORD2)) + '\n',
-            dumps(deepcopy(RECORD3)) + '\n'])
+        def result(a):
+            return File([
+                dumps(PARAMS) + '\n',
+                dumps(deepcopy(RECORD0)) + '\n',
+                dumps(deepcopy(RECORD1)) + '\n',
+                dumps(deepcopy(RECORD2)) + '\n',
+                dumps(deepcopy(RECORD3)) + '\n'])
 
         with patch.object(builtins, 'open') as mockMethod:
             mockMethod.side_effect = result
@@ -366,10 +369,14 @@ class TestDiamondReadsAlignments(TestCase):
         The adjustHspsForPlotting function must alter HSPs so that zero
         evalues are set randomly high.
         """
-        result = lambda a: File([
-            dumps(PARAMS) + '\n', dumps(deepcopy(RECORD0)) + '\n',
-            dumps(deepcopy(RECORD1)) + '\n', dumps(deepcopy(RECORD2)) + '\n',
-            dumps(deepcopy(RECORD3)) + '\n', dumps(deepcopy(RECORD4)) + '\n'])
+        def result(a):
+            return File([
+                dumps(PARAMS) + '\n',
+                dumps(deepcopy(RECORD0)) + '\n',
+                dumps(deepcopy(RECORD1)) + '\n',
+                dumps(deepcopy(RECORD2)) + '\n',
+                dumps(deepcopy(RECORD3)) + '\n',
+                dumps(deepcopy(RECORD4)) + '\n'])
 
         with patch.object(builtins, 'open') as mockMethod:
             mockMethod.side_effect = result

--- a/test/test_dna.py
+++ b/test/test_dna.py
@@ -265,7 +265,7 @@ class TestCompareDNAReads(TestCase):
 
     def testNonMatchingAmbiguityInFirst(self):
         """
-        Two sequences that match exactly, apart from one (incompatible) 
+        Two sequences that match exactly, apart from one (incompatible)
         ambiguity in the second sequence, must compare as expected.
         """
         self.assertEqual(

--- a/test/test_features.py
+++ b/test/test_features.py
@@ -155,7 +155,9 @@ class Test_FeatureList(TestCase):
         If the sequence fetcher returns C{None} we must be marked as being
         offline.
         """
-        fetcher = lambda title, db='database': None
+        def fetcher(title, db='database'):
+            return None
+
         featureList = FeatureList('title', 'database', set(),
                                   sequenceFetcher=fetcher)
         self.assertEqual(True, featureList.offline)
@@ -165,7 +167,9 @@ class Test_FeatureList(TestCase):
         If the sequence fetcher indicates we're offline, the feature list
         must have length zero.
         """
-        fetcher = lambda title, db='database': None
+        def fetcher(title, db='database'):
+            return None
+
         featureList = FeatureList('title', 'database', set(),
                                   sequenceFetcher=fetcher)
         self.assertEqual(0, len(featureList))
@@ -188,7 +192,9 @@ class Test_FeatureList(TestCase):
         If the sequence fetcher does not return C{None} we must be marked as
         being online.
         """
-        fetcher = lambda title, db='database': SeqRecord(None)
+        def fetcher(title, db='database'):
+            return SeqRecord(None)
+
         featureList = FeatureList('title', 'database', set(),
                                   sequenceFetcher=fetcher)
         self.assertEqual(False, featureList.offline)
@@ -198,7 +204,9 @@ class Test_FeatureList(TestCase):
         If the sequence fetcher returns a record with no features, the
         L{FeatureList} instance must have length zero.
         """
-        fetcher = lambda title, db='database': SeqRecord(None)
+        def fetcher(title, db='database'):
+            return SeqRecord(None)
+
         featureList = FeatureList('title', 'database', set(),
                                   sequenceFetcher=fetcher)
         self.assertEqual(0, len(featureList))
@@ -289,7 +297,9 @@ class Test_FeatureAdder(TestCase):
         """
         A L{_FeatureAdder} must set the title of its figure.
         """
-        fetcher = lambda title, db='database': None
+        def fetcher(title, db='database'):
+            return None
+
         featureAdder = _FeatureAdder()
         fig = plt.subplot(111)
         fig.set_title = MagicMock()
@@ -302,7 +312,9 @@ class Test_FeatureAdder(TestCase):
         """
         A L{_FeatureAdder} must set the title of its figure.
         """
-        fetcher = lambda title, db='database': None
+        def fetcher(title, db='database'):
+            return None
+
         featureAdder = _FeatureAdder()
         fig = plt.subplot(111)
         fig.set_yticks = MagicMock()
@@ -316,7 +328,9 @@ class Test_FeatureAdder(TestCase):
         the C{add} method of the L{_FeatureAdder} must return C{None}. The
         C{text} and C{axis} methods on the fig must both be called correctly.
         """
-        fetcher = lambda title, db='database': None
+        def fetcher(title, db='database'):
+            return None
+
         featureAdder = _FeatureAdder()
         fig = plt.subplot(111)
         fig.text = MagicMock()
@@ -334,7 +348,9 @@ class Test_FeatureAdder(TestCase):
         the C{text} and C{axis} methods on the fig must both be called
         correctly and the C{add} method must return C{[]}.
         """
-        fetcher = lambda title, db='database': SeqRecord(None)
+        def fetcher(title, db='database'):
+            return SeqRecord(None)
+
         featureAdder = _FeatureAdder()
         featureAdder.WANTED_TYPES = ('site',)
         fig = plt.subplot(111)
@@ -536,7 +552,10 @@ class TestNucleotideFeatureAdder(TestCase):
         fig.plot = MagicMock()
         fig.axis = MagicMock()
         fig.legend = MagicMock()
-        adjuster = lambda x: 3 * x
+
+        def adjuster(x):
+            return 3 * x
+
         result = featureAdder.add(fig, 'title', 0, 300, adjuster,
                                   sequenceFetcher=fetcher)
         fig.plot.assert_called_with(

